CRM/Event/ActionMapping.php

```diff
@@ -146,7 +146,7 @@ public function createQuery($schedule, $phase, $defaultParams) {
     }
 
     $query->join('r', 'INNER JOIN civicrm_event r ON e.event_id = r.id');
-    if ($schedule->recipient_listing && $schedule->limit_to) {
+    if ($schedule->recipient_listing && in_array($schedule->limit_to, [2, 3])) {
```

Comment:
Another in_array, which could just be && ($schedule->limit_to === 2 || $schedule->limit_to== 3])